SamplesDashboard/SamplesDashboard/Models/YamlHeader.cs

```diff
@@ -14,5 +21,121 @@ public class YamlHeader
+        internal bool NoDependenciesWasSpecified { get; set; }
+
+        public void MergeWith(YamlHeader mergeTarget)
+        {
+            if (mergeTarget == null) return;
+
+            Languages = Languages ?? mergeTarget.Languages;
+            Extensions = Extensions ?? mergeTarget.Extensions;
+            DependencyFile = DependencyFile ?? mergeTarget.DependencyFile;
+            NoDependencies = NoDependenciesWasSpecified ? NoDependencies : NoDependencies || mergeTarget.NoDependencies;
+        }
+
+        public static YamlHeader FromString(string headerContent)
+        {
+            var deserializer = new DeserializerBuilder()
+                    .WithNamingConvention(CamelCaseNamingConvention.Instance)
+                    .IgnoreUnmatchedProperties()
+                    .Build();
+            var header = deserializer.Deserialize<YamlHeader>(headerContent);
+
+            if (headerContent.Contains("noDependencies"))
+            {
+                header.NoDependenciesWasSpecified = true;
+            }
+
+            return header;
+        }
+
+        public static async Task<YamlHeader> GetFromRepo(HttpClient httpClient,
+                                                         string repoName,
+                                                         string branch)
+        {
+            var currentHeader = await GetFromReadme(httpClient, repoName, branch);
+            var repoNameFileHeader = await GetFromFile(httpClient, repoName, branch, $"{repoName}.yml");
+            if (repoNameFileHeader != null)
+            {
+                repoNameFileHeader.MergeWith(currentHeader);
+                currentHeader = repoNameFileHeader;
+            }
+
+            var devxFileHeader = await GetFromFile(httpClient, repoName, branch, "devx.yml");
+            if (devxFileHeader != null)
+            {
+                devxFileHeader.MergeWith(currentHeader);
+                currentHeader = devxFileHeader;
+            }
+
+            return currentHeader;
+
+        }
+
+        public string LanguagesString 
+        {
+            get
+            {
+                return Languages == null ? "" : string.Join(",", Languages);
+            }
+        }
+
+        public string ServicesString
+        {
+            get
+            {
+                return (Extensions == null || Extensions.Services == null) ? 
+                    "" : string.Join(",", Extensions.Services);
+            }
+        }
+
+        private static async Task<YamlHeader> GetFromReadme(HttpClient httpClient,
+                                                            string repoName,
+                                                            string branch)
+        {
+            var responseMessage = await httpClient.GetAsync(
+                $"https://raw.githubusercontent.com/microsoftgraph/{repoName}/{branch}/README.md");
+
+            if (responseMessage.StatusCode.ToString().Equals("NotFound"))
```

Comment:
```suggestion
            if ("NotFound".Equals(responseMessage.StatusCode.ToString(), StringComparison.OrdinalIgnoreCase))
```
inverting the condition protects you from null values, even if that's technically hard to get a null value in this case